tap_sftp/client.py

```diff
@@ -82,6 +84,44 @@ def match_files_for_table(self, files, table_name, search_pattern):
         matcher = re.compile(search_pattern)
         return [f for f in files if matcher.search(f["filepath"])]
 
+    def should_skip_zip_file(self, filename):
```

Comment:
Can we create a get_row_iterators fun in csv_helper and check this? So we no need to traverse zip again? 

```
def get_row_iterators(iterable, options={}, infer_compression=False):
    """Accepts an interable, options and a flag to infer compression and yields
    csv.DictReader objects which can be used to yield CSV rows."""
    if infer_compression:
        compressed_iterables = compression.infer(iterable, options.get('file_name'))

    for item in compressed_iterables:
        yield get_row_iterator(item, options=options)
```